### PR TITLE
feat: change test outputs to be html and add toggle buttons

### PR DESCRIPTION
**Pull request by civsiv**
Summary page:
Screenshot 2024-02-20 at 11 06 40 (2)

Test output page (on load, all but first error collapsed) :
Screenshot 2024-02-20 at 11 07 02

Open all:
Screenshot 2024-02-20 at 11 07 14

Collapse all:
Screenshot 2024-02-20 at 11 07 23

### CODE REVIEW

File: packages/openactive-integration-tests/test/report-templates/report.md.handlebars [hunk 1]
```diff
@@ -1,5 +1,20 @@
+<head>
+  <style>
```
Comment (civsiv) [inline packages/openactive-integration-tests/test/report-templates/report.md.handlebars:2]: Some basic CSS to make headers more obviously clickable

**Review by lukehesluke — CHANGES_REQUESTED** [review]
@civsiv apologies if I'm using it wrong but this is what I've found, just by looking at the output HTML files:

## Individual Test Page

* ✅ On page load, all but the first error are collapsed
* ✅  If I press "Toggle All Sections" the first time, all sections become un-collapsed. Then later presses toggle between all collapsed and all un-collapsed
* ✅ If all sections are un-collapsed, the "Collapse All But First Error" button does as expected.
* ❌ If all sections are already collapsed, the "Collapse All But First Error" button doesn't do anything
* ✅ It's possible to collapse a section by clicking on the section header, (which also has all the visual markers of being clickable)
* ❌ However, clicking on an already-collapsed section does not un-collapse it
* ❌  `< Return to Summary` link (in the top left) is currently broken - it links to summary.md

* Suggestion: Split the "Toggle All Sections" button into "Collapse all Sections" and "Show all Sections". Or maybe have some other way of indicating whether pressing the button will collapse or show? This is a random example but one I could think of which is: https://google.github.io/styleguide/lispguide.xml. They have the same button, but it uses an arrow to indicate whether it's showing or collapsing
* Minor Suggestion: Rename the "Collapse All But First Error" button into "Show Only First Error"
* Suggestion: Is there some tool/library that is easy to set up which can be used to make those code blocks a little more readable? Like syntax highlighting and maybe putting a box around them making it easy to copy all the code. Maybe hightlight.js?

## Summary page

* ❌ Make it possible to show/hide successful tests (maybe toggle button(s))?
* ✅ Everything else looks good

**Comment by nickevansuk** [comment]
Further to Luke's comments:
- CSS of H2s updated to include `cursor: pointer;`
- Include a summary tick or cross in the H2 line itself - to make it clear where the errors are

The second point is a little gnarly, so here's some thoughts of how to achieve this:
- Similar to `{{#logsFor . "spec"}}` include `{{ consoleValidationIcon (statusFor .) }}` (new function statusFor as a subexpression) inside `{{#each activeSuites }}` ([ref](https://github.com/openactive/openactive-test-suite/blob/374ba842c5793c7fba260846b6acd7824cad2427/packages/openactive-integration-tests/test/report-templates/report-cli.handlebars#L11))
- Create a new helper similar to `logsFor` called `statusFor` ([ref](https://github.com/openactive/openactive-test-suite/blob/374ba842c5793c7fba260846b6acd7824cad2427/packages/openactive-integration-tests/test/report-generator.js#L105))
- Implement a `statusFor` similar to `logsFor` in the logger ([ref](https://github.com/openactive/openactive-test-suite/blob/374ba842c5793c7fba260846b6acd7824cad2427/packages/openactive-integration-tests/test/helpers/logger.js#L447)).
- Use logic similar to `get overallStatus()` ([ref](https://github.com/openactive/openactive-test-suite/blob/374ba842c5793c7fba260846b6acd7824cad2427/packages/openactive-integration-tests/test/helpers/logger.js#L288)), but additionally filtering on the suite name. Ideally abstract the logic out so it can be used in both places. Note the inheritance structure here - `ReporterLogger extends BaseLogger` - `logsFor` is in `ReporterLogger`, `overallStatus` is in `BaseLogger`.

**Comment by civsiv** [comment]
New Zip: 
[output.zip](https://github.com/openactive/openactive-test-suite/files/14376293/output.zip)

**Comment by nickevansuk** [comment]
Suggest adding a thin border to the top of the H2:

```
h2 {
    cursor: pointer;
    border-top-color: #CCC;
    border-top-width: 1px;
    border-top-style: solid;
    padding-top: 5px;
}
```

**Review by lukehesluke — APPROVED** [review]

**Comment by civsiv** [comment]
Screenshot 2024-02-27 at 10 13 11
Screenshots are unaffected!

**Pull request merged**